src/Stratis.Features.FederatedPeg/Wallet/FederationWalletSyncManager.cs

```diff
@@ -144,6 +144,7 @@ private async Task OnProcessBlockAsync(Block block, CancellationToken cancellati
 
                     this.logger.LogInformation("Reorg detected, going back from '{0}' to '{1}'.", this.walletTip, fork);
 
+                    this.walletManager.RemoveTransientTransactions();
```

Comment:
As @quantumagi pointed out won't we need to remove also reorged trx?

Reply:
I think the Fed wallet is missing a similar functionality as this in the mainchain wallet `RefreshInputKeysLookupLock`